### PR TITLE
updated dependencies as far as possible

### PR DESCRIPTION
**Pull request by GottZ**
updated dependencies as far as possible

## Description
i just went through all dependencies and upgraded them as far as possible without required changes to the code base.

## Related Issue
none

## Motivation and Context
never let dependencies age.

## How Has This Been Tested?
`npm run lint`
`npm run test`
`npm start` → creating a new account → logging in
since there sadly is no tests i cannot ensure everything works.
please verify this before accepting this pull request.

environment:
latest chrome, mac os 10.12.6, node lts/carbon

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by GottZ** [comment]
if anyone is wondering about `ajv`… it's required for various packages. just remove it, do npm install and you'll see what i mean.

**Comment by GottZ** [comment]
so flood is RiP?

**Comment by jfurrow** [comment]
@GottZ No, but I'm busy, and this PR is not a priority.

File: package.json [hunk 1]
```diff
@@ -16,105 +16,106 @@
...
-    "babel-runtime": "^6.25.0",
+    "babel-loader": "^7.1.4",
+    "babel-plugin-transform-react-jsx": "^6.24.1",
+    "babel-plugin-transform-runtime": "^6.23.0",
+    "babel-polyfill": "^6.26.0",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react-app": "^3.1.1",
+    "babel-preset-stage-2": "^6.24.1",
+    "babel-runtime": "^6.26.0",
     "bcrypt": "^1.0.3",
```
Comment (noraj) [inline package.json]: I tried to build with the upstream git + this PR + "bcrypt": "^2.0.1", with nodejs 10.0.0 in order to address https://github.com/jfurrow/flood/issues/656

Comment (noraj) [reply package.json]: This should be a benign change https://github.com/kelektiv/node.bcrypt.js#migrating-from-v10x. I worked for me.

File: package.json [hunk 2]
```diff
@@ -16,105 +16,106 @@
     "start:development:client": "node client/scripts/start.js"
   },
   "dependencies": {
-    "argon2": "^0.16.2",
```
Comment (noraj) [inline package.json]: v0.17.0 introduce  `Major refactoring` but I tested with 0.17.3 and the authentication worked perfectly.

File: package.json [hunk 3]
```diff
@@ -16,105 +16,106 @@
...
-    "autoprefixer": "^7.1.2",
```
Comment (noraj) [inline package.json]: One should see if the changes introduced in those versions are impacting Flood

https://github.com/postcss/autoprefixer/releases/tag/8.4.0
https://github.com/postcss/autoprefixer/releases/tag/8.3.0
https://github.com/postcss/autoprefixer/releases/tag/8.2.0
https://github.com/postcss/autoprefixer/releases/tag/8.1.0
https://github.com/postcss/autoprefixer/releases/tag/8.0.0
https://github.com/postcss/autoprefixer/releases/tag/7.2.0

File: package.json [hunk 4]
```diff
@@ -16,105 +16,106 @@
...
-    "axios": "^0.16.2",
```
Comment (noraj) [inline package.json]: [0.17.0](https://github.com/axios/axios/blob/master/CHANGELOG.md#0170-oct-21-2017) introduced two breaking changes that may impact flood.

File: package.json [hunk 5]
```diff
@@ -16,105 +16,106 @@
...
-    "babel-core": "6.25.0",
```
Comment (noraj) [inline package.json]: [6.26.0](https://github.com/babel/babel/blob/master/CHANGELOG.md#6260-2017-08-16) doesn't introduce breaking changes

File: package.json [hunk 6]
```diff
@@ -16,105 +16,106 @@
...
-    "transform-loader": "^0.2.3",
-    "url-loader": "0.5.9",
+    "style-loader": "0.20.3",
+    "tar-stream": "^1.5.5",
+    "transform-loader": "^0.2.4",
+    "url-loader": "1.0.1",
     "webpack": "^3.4.1",
-    "xmlrpc": "^1.3.0"
+    "ajv": "^6.4.0",
+    "xmlrpc": "^1.3.2"
```
Comment (noraj) [inline package.json]: [1.3.2](https://github.com/baalexander/node-xmlrpc/releases/tag/v1.3.2) *Drops support CI/Support for node v0.8* but Flood already require node 7.x + so this is not a problem.

File: package.json [hunk 7]
```diff
@@ -16,105 +16,106 @@
...
-    "tar-stream": "^1.5.4",
-    "transform-loader": "^0.2.3",
-    "url-loader": "0.5.9",
+    "style-loader": "0.20.3",
+    "tar-stream": "^1.5.5",
+    "transform-loader": "^0.2.4",
+    "url-loader": "1.0.1",
     "webpack": "^3.4.1",
-    "xmlrpc": "^1.3.0"
+    "ajv": "^6.4.0",
```
Comment (noraj) [inline package.json]: This dependency should be put in the alphabetical order.

File: package.json [hunk 8]
```diff
@@ -16,105 +16,106 @@
...
+    "flux": "^3.1.3",
+    "fs-extra": "^5.0.0",
     "geoip-country-only": "^1.1.8",
-    "html-webpack-plugin": "^2.30.1",
+    "html-webpack-plugin": "^3.1.0",
     "inuit-page": "^0.2.1",
     "inuit-reset": "^0.1.1",
-    "joi": "^13.0.2",
-    "jsonwebtoken": "^7.0.1",
+    "joi": "^13.1.2",
```
Comment (noraj) [inline package.json]: No breaking change.

File: package.json (hunk 7 above)
Comment (jfurrow) [inline package.json]: @GottZ I removed `ajv` and everything compiled as expected in node 8 and 9. Please remove.

**Comment by jfurrow** [comment]
@noraj Thanks for reviewing each dependency. Honestly I was just planning to manually Flood's functionality, but I really appreciate your thorough review.

**Comment by boistordu** [comment]
@noraj  thanks didn't have the time yet to have done much but I'm working on it right now

**Comment by jfurrow** [comment]
Thanks @GottZ! I'm closing in favor of https://github.com/jfurrow/flood/pull/665, but I grabbed your commit so you'll still get the proper contribution credit.

**Pull request closed without merging**